src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(todosList => {
+      setTodos(todosList);
+      setInitialTodos(todosList);
+    });
+  }, []);
+
+  const handleEyeClick = useCallback(
+    (selectedTodoId: number) => {
+      setSelectedTodo(todos.find(todo => todo.id === selectedTodoId) || null);
+    },
+    [todos],
+  );
+
+  const handleFiltrationQueries = useCallback(
+    (finishQuery = 'all', searchQuery = '') => {
+      setTodos(
+        initialTodos.filter(todo => {
+          if (finishQuery === 'all') {
+            return todo.title.toLowerCase().includes(searchQuery.toLowerCase());
+          }
+
+          return (
+            todo.title.toLowerCase().includes(searchQuery.toLowerCase()) &&
+            todo.completed === (finishQuery === 'active' ? false : true)
+          );
+        }),
+      );
+    },
```

Comment:
- Filtered todos is a dynamic value that depends on selected filters so don't store it in the state, just calculate
- Create a helper function to filter and move it to utils/helpers file/folder
- Get rid of magic values such as "all"/"active"

Simple example:
```typescript
const filteredTodos = filterTodos(todos, { query, status });
```